internal/config/cgmgr/systemd.go

```diff
@@ -77,18 +81,100 @@ func (*SystemdManager) MoveConmonToCgroup(cid, cgroupParent, conmonCgroup string
 	// Set the systemd KillSignal to SIGPIPE that conmon ignores.
 	// This helps during node shutdown so that conmon waits for the container
 	// to exit and doesn't forward the SIGTERM that it gets.
-	killSignalProp := systemdDbus.Property{
-		Name:  "KillSignal",
-		Value: dbus.MakeVariant(int(unix.SIGPIPE)),
+	props := []systemdDbus.Property{
+		{
+			Name:  "KillSignal",
+			Value: dbus.MakeVariant(int(unix.SIGPIPE)),
+		},
+		systemdDbus.PropAfter("crio.service"),
 	}
+
+	if resources.CPU != nil {
+		if resources.CPU.Cpus != "" {
+			bits, err := rangeToBits(resources.CPU.Cpus)
+			if err != nil {
+				return "", errors.Wrapf(err, "cpuset conversion error")
+			}
+			props = append(props, systemdDbus.Property{
+				Name:  "AllowedCPUs",
+				Value: dbus.MakeVariant(bits),
+			})
+		}
+		if resources.CPU.Shares != nil {
+			props = append(props, systemdDbus.Property{
+				Name:  "CPUShares",
+				Value: dbus.MakeVariant(resources.CPU.Shares),
+			})
+		}
+	}
+
 	logrus.Debugf("Running conmon under slice %s and unitName %s", cgroupParent, conmonUnitName)
-	if err := utils.RunUnderSystemdScope(pid, cgroupParent, conmonUnitName, killSignalProp, systemdDbus.PropAfter("crio.service")); err != nil {
+	if err := utils.RunUnderSystemdScope(pid, cgroupParent, conmonUnitName, props...); err != nil {
```

Comment:
Wonder if it's possible to reuse runc systemd manager's Apply()/Set() here. It does a job similar to `utils.RunUnderSystemdScope`, can be used to set arbitrary properties (via `cgroups.SystemdProps`), and in general does a better job.

Reply:
unfortunatley Apply/Set aren't yet a very portable API (from what I was able to put together). Here's what I had to do to use them for this case: https://github.com/haircommander/cri-o/commit/82b09974a31c4e9c273de980592c8f1feb65c623 . I would believe I haven't used the API correctly, but I think I'd like to port this to using runc at a later time (if ever)